implementations/python/mzlib/backends/base.py

```diff
@@ -356,21 +319,35 @@ class SpectralLibraryWriterBase(object, metaclass=SubclassRegisteringMetaclass):
     def __init__(self, filename, **kwargs):
         self.filename = filename
 
+    def _filter_attributes(self, attributes: Attributed, filter_fn: Callable) -> Iterable:
+        if isinstance(attributes, AttributedEntity):
+            attributes = attributes.attributes
+        for attrib in attributes:
+            if filter_fn(attrib):
+                yield attrib
+
+    def _not_analyte_mixture_term(self, attrib):
+        if attrib:
```

Comment:
`return not attrib or attrib[0] != ANALYTE_MIXTURE_TERM`
Doesn't really make a difference.